sys/cheri/cheri.h

```diff
@@ -62,12 +62,8 @@ struct cheri_object {
  * pointer should (presumably) be relative to the $ddc/$csp defined here.
  */
 struct cheri_signal {
```

Comment:
This belongs in sys/mips if you're doing cleanup

Reply:
Well, one of the questions of this series is if we still want `struct cheri_signal` at all or if I should add additional commits to remove it entirely.  Certainly it seems to have rotted quite a bit.  The fact that hybrid doesn't reset idc when sandboxed is also somewhat odd perhaps?

In general sendsig() leaks register values from the interrupted context except for the ones that are explicitly overwritten.  It would arguably be cleaner if after saving the signal frame we zeroed everything in the trap frame and only set the registers needed to run the signal handler (in general, not even just for CHERI).

Reply:
Maybe axing it and libcheri at the same time would make sense, then we can start fresh for the modern sandboxing frameworks.